src/engine/Server.cpp

```diff
@@ -20,22 +20,22 @@ template <typename T>
 using Awaitable = Server::Awaitable<T>;
 
 // __________________________________________________________________________
-Server::Server(const int port, const int numThreads, size_t maxMemGB,
-               std::string accessToken)
+Server::Server(unsigned short port, int numThreads, size_t maxMemGB,
+               std::string&& accessToken, bool usePatternTrick)
     : _numThreads(numThreads),
       _port(port),
-      accessToken_(accessToken),
-      _allocator{ad_utility::makeAllocationMemoryLeftThreadsafeObject(
-                     maxMemGB * (1ull << 30u)),
-                 [this](size_t numBytesToAllocate) {
-                   _cache.makeRoomAsMuchAsPossible(
-                       static_cast<double>(numBytesToAllocate) / sizeof(Id) *
-                       MAKE_ROOM_SLACK_FACTOR);
-                 }},
+      accessToken_(std::move(accessToken)),
+      _allocator{
+          ad_utility::makeAllocationMemoryLeftThreadsafeObject(maxMemGB *
+                                                               (1ull << 30u)),
+          [this](size_t numBytesToAllocate) {
+            _cache.makeRoomAsMuchAsPossible(MAKE_ROOM_SLACK_FACTOR *
+                                            numBytesToAllocate / sizeof(Id));
+          }},
       _sortPerformanceEstimator(),
       _index(),
       _engine(),
-      _initialized(false),
+      _enablePatternTrick(usePatternTrick),
```

Comment:
Now the configuration of the patter stuff is in two places. That is also not nice. Maybe we can check whether we need this `enablePatternTrick` thing at all.